MobileCenterPush/MobileCenterPush/MSPush.m

```diff
@@ -38,10 +38,19 @@
 static NSString *const kMSPushNotificationCustomDataKey = @"mobile_center";
 
 /**
+ * Key for NSUserNotificationCenter delegate property.
+ */
+static NSString *const kMSUserNotificationCenterDelegateKey = @"delegate";
+
+/**
  * Singleton
  */
 static MSPush *sharedInstance = nil;
 static dispatch_once_t onceToken;
+#if TARGET_OS_OSX
+static id<NSUserNotificationCenterDelegate> userNotificationCenterDelegate;
+static void *UserNotificationCenterDelegateContext = &UserNotificationCenterDelegateContext;
```

Comment:
Do we need this? Looks like push doesn't have a super or child class to conflict to.

Reply:
A kind of safeguard. We only intercept if it is our context.

Reply:
All right.